File: lec02/01_selection_sort/selectionSort.py
# selectionSort.py
def selection_sort(arr):
  """선택 정렬 함수"""
  n = len(arr)

  for i in range(n - 1):
    min_index = i

    # i번째 위치부터 끝까지 최솟값 찾기
    for j in range(i + 1, n):
      if arr[j] < arr[min_index]:
        min_index = j

    # 최솟값을 i번째 위치로 교환
    if min_index != i:
      arr[i], arr[min_index] = arr[min_index], arr[i]


# 사용 예시
if __name__ == "__main__":
  arr = [64, 34, 25, 12, 22, 11, 90]

  print("정렬 전 배열:", arr)

  selection_sort(arr)

  print("정렬 후 배열:", arr)

File: lec02/01_selection_sort/selectionSort.c
// selectionSort.c
#include <stdio.h>

// 선택 정렬 함수
void selectionSort(int arr[], int size) {
  for (int i = 0; i < size - 1; i++) {
    int minIndex = i;

    // i번째 위치부터 끝까지 최솟값 찾기
    for (int j = i + 1; j < size; j++) {
      if (arr[j] < arr[minIndex]) {
        minIndex = j;
      }
    }

    // 최솟값을 i번째 위치로 교환
    if (minIndex != i) {
      int temp = arr[i];
      arr[i] = arr[minIndex];
      arr[minIndex] = temp;
    }
  }
}

// 배열 출력 함수
void printArray(int arr[], int size) {
  for (int i = 0; i < size; i++) {
    printf("%d ", arr[i]);
  }
  printf("\n");
}

// 사용 예시
int main() {
  int arr[] = {64, 34, 25, 12, 22, 11, 90};
  int size = sizeof(arr) / sizeof(arr[0]);

  printf("정렬 전 배열: ");
  printArray(arr, size);

  selectionSort(arr, size);

  printf("정렬 후 배열: ");
  printArray(arr, size);

  return 0;
}
